fpga/greg-util.py

```diff
@@ -157,6 +157,25 @@ def init_gui(self):
 
         self.bind('<Control-V>', self.textbox_write.focus)
 
+    def bswap_bytes(self, buf):
+        """ cuts to 16-bit and swaps BIG<->LITTLE from buffer byte array """
+        # FPGA Sends and Receives in big endian
+        # ARM translates the i2c address
+        # ARM does not translate the data
+        buf = buf[0:2]
+        val = int.from_bytes(buf, "big")
+        return val
+    
+    def bswap_int(self, val):
+        """ cuts to 16-bit and swaps BIG<->LITTLE from int """
+        # FPGA Sends and Receives in big endian
+        # ARM translates the i2c address
+        # ARM does not translate the data
+        val = val.to_bytes(4, "little")
+        val = val[0:2]
+        val = int.from_bytes(val, "big")
```

Comment:
converting from little and then from big is the same as just converting from little, this part makes the code a bit confusing. Can it be rewritten to only use to_bytes/from_bytes once?

Reply:
Since the function btn_write_clicked previously passed an int type to be written with the ctrl_transfer function I wanted to keep the same type after swapping the byte order. That means converting from the supplied int to a byte array, swapping the order, then converting back to an int. I suppose you could choose which order to do the endianess swapping in if you took a different slice of the byte array, but I think its clear to have a swap to a byte array and then back to an int in the bswap_int function. The bswap_bytes function has a byte array passed in so there is no need to convert before swapping the order.